README.md

```diff
@@ -190,15 +196,23 @@ Scheduled jobs transition to "AVAILABLE" after the scheduled time has passed.
 
 Jobs that are intended to run immediately start in an "AVAILABLE" state.
 
-The producer will update a batch of jobs setting the state to "RESERVED", with an expiry of 5 minutes.
+The producer will update a batch of jobs setting the state to "RESERVED", with an expiry of 5 minutes unless otherwise configured.
 
-Once a consumer is given a job, it increments the attempt counter and updates the state to "IN_PROGRESS", with an expiry of 5 minutes.
-If the job is being retried, the expiry will be 5 minutes * the attempt counter.
+Once a consumer is given a job, it increments the attempt counter and updates the state to "IN_PROGRESS", with an initial expiry of the configurable as `base_expiry_seconds`, which defaults to 5 minutes.
+If the job is being retried, the expiry will be base expiry * the attempt counter.
 
-If successful, the consumer will delete the job from the queue.
-If unsuccessful, the job remains in the "IN_PROGRESS" state until it expires.
+If successful, the consumer can delete the job from the queue using the preloaded multi passed to the `perform/2` job handler.
+If an exception is raised in the worker or a successful processing attempt failes to successfully comitt the preloaded multi, the job is not deleted and remains in the "IN_PROGRESS" state until it expires.
 
 Jobs in the "RESERVED" or "IN_PROGRESS" state past the expiry time will be returned to the "AVAILABLE" state.
 
 Expired jobs in the "IN_PROGRESS" state with attempts >= MAX_ATTEMPTS move to a "FAILED" state.
 Failed jobs are kept in the database so that application developers can handle the failure.
+
+## Job Timeouts and Transactional Safety
+
+When performing long-running jobs or when configuring a low expiry timout, keep in mind that a job may be retried before it has finished and the retry has no proactive mechanism to cancel the running job.
+
+In the case that the initial job attempts to finish and commit a result, and the commit includes the preloaded multi passed as the first parameter to `perform/2`, the optimistic lock will fail the transaction.
+
+In the case where the job performs other side effects outside of the transaction such as calls to external APIs or additional database writes, these are suggested to implment other idempotency guarantees, as they will not be rolled back in a failed or duplicated job.
```

Comment:
Let me know if this sounds correct and helpful

Reply:
This is a great description of how it works today. 👍 
I'm sure the mechanism could be improved. Eg, some way for a job to signal that it is still alive and the expiry should be extended.

Reply:
implment -> implement